core/trino-main/src/test/java/io/trino/execution/scheduler/TestPartitionedPipelinedOutputBufferManager.java

```diff
@@ -47,10 +47,9 @@ public void test()
                 .hasMessage("Unexpected new output buffer 5");
         assertOutputBuffers(hashOutputBufferManager.getOutputBuffers());
 
-        // try to a buffer out side of the partition range, which should result in an error
-        assertThatThrownBy(() -> hashOutputBufferManager.addOutputBuffer(new OutputBufferId(6)))
-                .isInstanceOf(IllegalStateException.class)
-                .hasMessage("Unexpected new output buffer 6");
+        // try to set no more buffers again, which should not result in an error
+        // and output buffers should not change
+        hashOutputBufferManager.noMoreBuffers();
```

Comment:
```
        assertThatThrownBy(() -> hashOutputBufferManager.addOutputBuffers(ImmutableList.of(new OutputBufferId(5)), false))
                .isInstanceOf(IllegalStateException.class)
                .hasMessage("Unexpected new output buffer 5");
        assertOutputBuffers(outputBufferTarget.get());

        assertThatThrownBy(() -> hashOutputBufferManager.addOutputBuffers(ImmutableList.of(new OutputBufferId(6)), true))
                .isInstanceOf(IllegalStateException.class)
                .hasMessage("Unexpected new output buffer 6");
        assertOutputBuffers(outputBufferTarget.get());
```

This is the logic before b4f40ab9. IUUC, the secondly `hashOutputBufferManager.addOutputBuffers` is aimed to check `noMoreBuffers`. Otherwise, we will try to a buffer out side of the partition range twice, which seems to have little substantive significance?

Reply:
BTW, I have checked the earlier version of this test case(05f904c6), and its logic is as follows:
```
// try to add another buffer, which should not result in an error and output buffers should not change
hashOutputBufferManager.addOutputBuffers(ImmutableList.of(new OutputBufferId(5)), false);
assertEquals(outputBuffers, outputBufferTarget.get());

// try to set no more buffers again, which should not result in an error and output buffers should not change
hashOutputBufferManager.addOutputBuffers(ImmutableList.of(new OutputBufferId(6)), true);
assertEquals(outputBuffers, outputBufferTarget.get());
```